picard/browser/browser.py

```diff
@@ -211,6 +213,26 @@ def _add_release(self, args):
         else:
             self._response(400, 'Missing parameter "token".')
 
+    def _auth(self, args):
+        if 'code' in args and args['code']:
+            tagger = QtCore.QCoreApplication.instance()
+            oauth_manager = tagger.webservice.oauth_manager
+            try:
+                state = args.get('state', [''])[0]
+                callback = oauth_manager.verify_state(state)
+            except OAuthInvalidStateError:
+                self._response(400, 'Invalid "state" parameter.')
+                return
+            to_main(
+                oauth_manager.exchange_authorization_code,
+                authorization_code=args['code'][0],
+                scopes='profile tag rating collection submit_isrc submit_barcode',
+                callback=callback,
+            )
+            self._response(200, "Authentication successful, you can close this window now.", 'text/html')
```

Comment:
Since this response is displayed in browser, it might be good to indicate it's Picard. But this can be improved in another PR.

Reply:
We could also generally improve the output by Picard a bit. There is not many user visible output in browser by Picard, but I think it would be useful and not much work to show a bit nicer pages with at least some minimal layout :)